extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java

```diff
@@ -88,19 +82,20 @@ void build(BuildProducer<AdditionalBeanBuildItem> additionalBeanProducer,
                 .collect(Collectors.toSet());
         new FlywayDatasourceBeanGenerator(dataSourceNames, generatedBeanBuildItem).createFlywayProducerBean();
 
-        registerNativeImageResources(resourceProducer, generatedResourceProducer,
-                discoverApplicationMigrations(getMigrationLocations(dataSourceNames)));
+        List<String> applicationMigrations = discoverApplicationMigrations(getMigrationLocations(dataSourceNames));
+        recorder.setApplicationMigrationFiles(applicationMigrations);
```

Comment:
It might be a silly idea but shouldn't we have a `Map<Datasource name, Migration files>`?

Reply:
I see the idea but then we do not have the `Datasource name` during scanning so we can't exploit the Map as we'd have liked.